packages/@react-native-windows/cli/src/e2etest/projectConfig.test.ts

```diff
@@ -101,6 +102,71 @@ const projects: TargetProject[] = [
       null,
     );
   }),
+  project('WithWinUI3', async (folder: string) => {
+    const windowsDir = path.join(folder, 'windows');
+    await tryMkdir(windowsDir);
+
+    const replacements = {
+      name: 'WithWinUI3',
+      namespace: 'WithWinUI3',
+      useMustache: true,
+      projectGuidUpper: testProjectGuid,
+      projectGuidLower: testProjectGuid.toLowerCase(),
+      useWinUI3: false,
+      useHermes: false,
+      packagesConfigCppNugetPackages: [
+        {
+          id: 'Microsoft.ReactNative.Cxx',
+          version: '1.0.0',
+          hasProps: false,
+          hasTargets: true,
+        },
+        {
+          id: 'Microsoft.UI.Xaml',
+          version: '2.3.4.5',
+          hasProps: false, // WinUI/MUX props and targets get handled by RNW's WinUI.props.
+          hasTargets: false,
+        },
+        {
+          id: 'Microsoft.WinUI',
+          version: '3.2.1.0',
+          hasProps: false, // WinUI/MUX props and targets get handled by RNW's WinUI.props.
+          hasTargets: false,
+        },
+      ],
+    };
+
+    await copyAndReplace(
+      path.join(templateRoot, 'cpp-app/proj/MyApp.sln'),
+      path.join(windowsDir, 'WithWinUI3.sln'),
```

Comment:
@NickGerleman/@asklar/@jonthyssel: Should we consider the same pattern for nuget for community modules?

Reply:
And hopefully make autolink aware of this...
@asklar: Similar for community modules that support both winui2/winui3

Reply:
@dannyvv could you clarify what you mean? this file is a test